src/components/LoginPage.component.js

```diff
@@ -27,20 +28,21 @@ export default class LoginPage extends Component {
     this.setState({ error: '' })
     const { username, password } = this.state
 
-    // RIPerino
-    // TODO: we know what we gotta do
-    if (username === 'admin' && password === 'admin') {
-      this.props.login()
-      this.props.history.push(ROUTES.INDEX)
-    } else {
-      this.setState({ error: 'incorrect credentials' })
-    }
+    this.props
+      .login({
+        username,
+        password
+      })
+      .then(() => this.props.history.push(ROUTES.INDEX))
   }
 
   render() {
     const { username, password, error } = this.state
+    const { isLoggedIn } = this.props
 
-    return (
+    return isLoggedIn ? (
```

Comment:
i think there should be a way to "protect" the route using react-router stuff in `index.js`

Reply:
instead of needing to explicitly pass `isLoggedIn` to every component